easybuild/easyblocks/q/quantumespresso.py

```diff
@@ -61,132 +63,269 @@ def __init__(self, *args, **kwargs):
 
-        if self.cfg['with_scalapack']:
-            dflags.append(" -D__SCALAPACK")
-            if self.toolchain.options.get('usempi', None):
-                if get_software_root("impi") and get_software_root("imkl"):
+        if comp_fam == toolchain.INTELCOMP:
+            if get_software_root("impi") and get_software_root("imkl"):
+                if LooseVersion(self.version) >= LooseVersion("6.2"):
+                    self.cfg.update('configopts', '--with-scalapack=intel')
+                elif LooseVersion(self.version) >= LooseVersion("5.1.1"):
                     self.cfg.update('configopts', '--with-scalapack=intel')
+                    self._repls += [
+                        ('SCALAPACK_LIBS', os.getenv('LIBSCALAPACK'), False)
+                    ]
+                elif LooseVersion(self.version) >= LooseVersion("5.0"):
+                    self.cfg.update('configopts', '--with-scalapack=yes')
+                self._dflags += ["-D__SCALAPACK"]
+        elif comp_fam == toolchain.GCC:
+            if get_software_root("OpenMPI") and get_software_root("ScaLAPACK"):
+                self.cfg.update('configopts', '--with-scalapack=yes')
+                self._dflags += ["-D__SCALAPACK"]
         else:
             self.cfg.update('configopts', '--without-scalapack')
 
+    def _add_libxc(self):
+        """Add libxc support to the build."""
         libxc = get_software_root("libxc")
-        if libxc:
-            libxc_v = get_software_version("libxc")
-            if LooseVersion(libxc_v) < LooseVersion("3.0.1"):
-                raise EasyBuildError("Must use libxc >= 3.0.1")
-            dflags.append(" -D__LIBXC")
-            repls.append(('IFLAGS', '-I%s' % os.path.join(libxc, 'include'), True))
-            if LooseVersion(self.version) < LooseVersion("6.5"):
-                extra_libs.append(" -lxcf90 -lxc")
-            else:
-                extra_libs.append(" -lxcf90 -lxcf03 -lxc")
+        if not libxc:
+            return
+
+        libxc_v = get_software_version("libxc")
+        if LooseVersion(libxc_v) < LooseVersion("3.0.1"):
+            raise EasyBuildError("Must use libxc >= 3.0.1")
+        if LooseVersion(self.version) >= LooseVersion("7.0"):
+            if LooseVersion(libxc_v) < LooseVersion("6.0.0"):
+                raise EasyBuildError("libxc support for QuantumESPRESSO 7.0 only available for libxc >= 6.0.0")
+            self.cfg.update('configopts', '--with-libxc=yes')
+            self.cfg.update('configopts', f'--with-libxc-prefix={libxc}')
+        elif LooseVersion(self.version) >= LooseVersion("6.4"):
+            if LooseVersion(libxc_v) >= LooseVersion("5.0"):
+                raise EasyBuildError("libxc support for QuantumESPRESSO 6.x only available for libxc <= 4.3.4")
+            self.cfg.update('configopts', '--with-libxc=yes')
+            self.cfg.update('configopts', f'--with-libxc-prefix={libxc}')
+        else:
+            self.extra_libs.append(" -lxcf90 -lxc")
```

Comment:
```suggestion
            self.extra_libs.append(" -lxcf90 -lxc ")
```
little more forgiving for future developers

Reply:
Changed to 
`self.extra_libs += ['-lxcf90', '-lxc']`
to let the `' '.join( ...)` afterward take care of it